python/pyiceberg/table/__init__.py

```diff
@@ -90,3 +110,16 @@ def snapshot_by_name(self, name: str) -> Optional[Snapshot]:
     def history(self) -> List[SnapshotLogEntry]:
         """Get the snapshot history of this table."""
         return self.metadata.snapshot_log
+
+    @cached_property
+    def io(self) -> FileIO:
+        return load_file_io({**self.catalog_properties, **self.metadata.properties, **self.config})
```

Comment:
Hey Ryan, that makes sense. My main reason to pass it to the Table is that it can potentially refresh the FileIO, but we could also just refresh the table. For example, when a credential gets expired. I've posted a follow up PR: https://github.com/apache/iceberg/pull/6170